src/java.base/share/classes/java/lang/invoke/GenerateJLIClassesHelper.java

```diff
@@ -315,7 +322,15 @@ static Map<String, byte[]> generateHolderClasses(Stream<String> traces)  {
                                         "linkToCallSite".equals(parts[2])) {
                                     builder.addCallSiteType(methodType);
                                 } else {
-                                    builder.addInvokerType(methodType);
+                                    MethodType mt = HolderClassBuilder.asMethodType(methodType);
+                                    if (HolderClassBuilder.checkInvokerTypeParams(mt)) {
+                                        builder.addInvokerType(methodType);
+                                    } else {
+                                        if (CDS.isDumpingArchive()) {
+                                            PlatformLogger.getLogger("java.lang.invoke")
+                                                    .warning("Invalid LF_RESOLVE " + parts[1] + " " + parts[2] + " " + parts[3]);
+                                        }
+                                    }
```

Comment:
The underlying cause of the failure in CDS is [JDK-8327499](https://bugs.openjdk.org/browse/JDK-8327499) -- MethodHandleStatics.traceLambdaForm() includes methods that cannot be handle by GenerateJLIClassesHelper.java.

Therefore, I think the warning should always be printed (e.g., when jdk.tools.jlink.internal.plugins.GenerateJLIClassesPlugin is used during the JDK build).

Also, maybe add a comment above line 326: 

```
// Work around JDK-8327499
```

Reply:
Fixed.